### PR TITLE
cephtool: pretty up error outputting during connect to cluster

### PR DESCRIPTION
**Pull request by jecluis**
The 'ceph' tool is awful conveying errors. This is an attempt at fixing
some of it, albeit poor at it.

Although this has been tested, proper unit tests still need to be added
to the pull request.

Reviews appreciated.

`Signed-off-by: Joao Eduardo Luis <joao@suse.de>`

### CODE REVIEW

**Comment by jecluis** [comment]
errr... while I do have a few tests to make sure we get prettier error messages with this branch, I'm not sure if I should actually get them upstream. These tests were meant as an automation until I got to a place I was rather not-as-unhappy, instead of having to type the same thing over and over again.

I don't really fancy the idea of having tests which success depends on the error **message**, as these are easily mutable and for good reason. Some of these also entail creating the appropriate error conditions, which are kind of overkill for unit tests and not relevant enough to warrant a workunit/qa-suite test of its own. Thoughts?

**Comment by jcsp** [comment]
Maybe you could have tests that check the output is a single line, or that it is not a python backtrace?

**Comment by alfredodeza** [comment]
What I've done for error messages that can be slightly different, is to assert the parts that I know should be present. Instead of asserting the whole line, or the whole output, I would do something like this:

    assert "permission denied" in error_message

When testing specifically about Permission Denied errors

File: src/ceph.in [hunk 1]
```diff
@@ -712,7 +712,7 @@ def main():
         return 1
     except Exception as e:
         print >> sys.stderr, 'Error connecting to cluster: {0}'.\
-            format(e.__class__.__name__)
+             format(e.message)
```
Comment (alfredodeza) [inline src/ceph.in]: this is a bit dangerous because `.message` is going away soon (it is deprecated). Only `.args` is ensured to be kept around.

Comment (alfredodeza) [inline src/ceph.in]: Running it with Python 2.6 would get you a deprecation warning:

    DeprecationWarning: BaseException.message has been deprecated as of Python 2.6

**Comment by liewegas** [comment]
@jecluis ping

needs rebase, at least

**Comment by jecluis** [comment]
@liewegas closing this pull request. Been a while since I looked at this. Will reopen if I believe it's in good shape to make the cut.

**Pull request closed without merging**